packages/gravity/kube-testing/src/spec/replication.ts

```diff
@@ -0,0 +1,351 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import { asyncTimeout, sleep, scheduleTaskInterval } from '@dxos/async';
+import { cancelWithContext, Context } from '@dxos/context';
+import { FeedFactory, FeedStore, type FeedWrapper } from '@dxos/feed-store';
+import { Keyring } from '@dxos/keyring';
+import { PublicKey } from '@dxos/keys';
+import { log } from '@dxos/log';
+import { TestBuilder as NetworkManagerTestBuilder, TestSwarmConnection } from '@dxos/network-manager/testing';
+import { createStorage, StorageType } from '@dxos/random-access-storage';
+import { ReplicatorExtension } from '@dxos/teleport-extension-replicator';
+import { range } from '@dxos/util';
+
+import { TestBuilder as SignalTestBuilder } from '../test-builder';
+import { PlanResults, TestParams, TestPlan, AgentEnv } from '../plan';
+import { generateKeyPair, TestKeyPair } from '../util';
+
+const REPLICATOR_EXTENSION_NAME = 'replicator';
+
+type FeedConfig = {
+  feedKey: string;
+  writable: boolean;
+};
+
+export type ReplicationTestSpec = {
+  agents: number;
+  swarmsPerAgent: number;
+  duration: number;
+
+  transport: 'webrtc' | 'webrtc-proxy';
+
+  targetSwarmTimeout: number;
+  fullSwarmTimeout: number;
+
+  feedsPerSwarm: number;
+  feedLoadInterval: number;
+  feedLoadChunkSize: number;
+
+  repeatInterval: number;
+
+  signalArguments: string[];
+};
+
+export type ReplicationAgentConfig = {
+  agentIdx: number;
+  signalUrl: string;
+  swarmTopicIds: string[];
+  feeds: Record<string, FeedConfig[]>;
+  feedKeys: TestKeyPair[];
+};
+
+export class ReplicationTestPlan implements TestPlan<ReplicationTestSpec, ReplicationAgentConfig> {
```

Comment:
Comment:
The purpose of this test is to...
Mention teleport, networkmanager, feedstore, etc.
What does it measure?